src/main/java/org/javarosa/core/services/Logger.java

```diff
@@ -81,5 +81,9 @@ public void run() {
         }
         throw crashException;
     }
+
+    public static void clear() {
+        logger = null;
```

Comment:
Just a quick note that "clear()" sounds to me like it clears the *log entries* themselves rather than the statically registered logger.

Might be worth renaming to something like "detachLogger" or something similar to make the outcome more clear